sdk/core/src/errors.rs

```diff
@@ -18,8 +18,10 @@ pub enum ParsingError {
 
 #[derive(Debug, Clone, PartialEq, thiserror::Error)]
 pub enum ParseError {
-    #[error("Split not found")]
-    SplitNotFound,
+    #[error("Expected token {} not found", 0)]
+    TokenNotFound(String),
+    #[error("Split {} not found", 0)]
```

Comment:
nit: should the token be printed inside of `'` characters? It might make it easier to read.